web/src/stream_create.js

```diff
@@ -280,8 +280,7 @@ function create_stream() {
             // The rest of the work is done via the subscribe event we will get
         },
         error(xhr) {
-            const msg = xhr.responseJSON.msg;
-            if (msg.includes("access")) {
+            if (xhr.responseJSON?.msg?.includes("access")) {
```

Comment:
The comments below make clear we should be changing this to look at the error codes; might be reasonable to just open an issue for that as a follow-up if it's not convenient to fix now.

Reply:
Sure.  Should we be using `ErrorCode.UNAUTHORIZED` for this?  We currently use it to mean "you're not authenticated" which is a little unfortunate, but this is really the right use case for it.  I guess to not muddle the technically-incorrect-current-usage, we should make a new `ErrorCode.UNAUTHORIZED_STREAM` or something.

I'm not sure we have a coherent policy about if we're trying to have mildly generic error codes, or if we expect each one to really only be raised in one specific place.

Reply:
We definitely don't have a clear policy, but in general have been favoring doing unique codes for each code we think a client might need to act on. I agree "request is not authenticated" should ideally be different from "you're not authorized", though it gets a bit messy with spectators requests, and we might want to think about how to manage that better.

Yeah I'm not sure, maybe a good "api design" conversation on how we want to handle this.